src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -166,15 +167,65 @@ public void EmitDebugLoc(int nativeOffset, int fileId, int linueNumber, int colN
         }
 
         [DllImport(NativeObjectWriterFileName)]
-        private static extern void FlushDebugLocs(IntPtr objWriter, string methodName, int methodSize);
-        public void FlushDebugLocs(int methodSize)
+        private static extern void EmitDebugVar(IntPtr objWriter, UInt32 varNumber, string name, UInt32 typeIndex, bool isParam, byte[] ranges, Int32 rangeCount);
+
+        public static byte[] GetBytes(NativeVarInfo[] nativeVarInfos)
+        {
+            int unitSize;
+            unsafe { unitSize = sizeof(NativeVarInfo); };
+            int len = nativeVarInfos.Length;
+            int size = unitSize * len;
+            byte[] bytes = new byte[size];
+
+            IntPtr ptr = Marshal.AllocHGlobal(unitSize);
+            for (int i = 0; i < len; i++)
+            {
+                Marshal.StructureToPtr(nativeVarInfos[i], ptr, true);
+                Marshal.Copy(ptr, bytes, unitSize * i, unitSize);
```

Comment:
This APIs are super slow. You should not need to be doing any of this. You can just pass the NativeVarInfo[] array directly to the PInvoke and it will just work (no unnecessary copies, etc.) because of it is array of blittable types.